### PR TITLE
RATIS-243 purge commit logs after persisting snapshots

### PR DESCRIPTION
**Pull request by andy2git**

### CODE REVIEW

**Comment by andy2git** [comment]
@szetszwo Finally I am good to go. You mentioned some logic here:
`SegmentedRaftLog#syncWithSnapshot`. I was wondering if my approach was the right direction.

**Comment by szetszwo** [comment]
Thanks @andy2git ,  I am reviewing your patch.  Sorry that I missed this pull request earlier.

**Comment by andy2git** [comment]
Thanks @szetszwo! Addressed code review comments.

File: ratis-server/src/main/java/org/apache/ratis/server/storage/RaftLog.java [hunk 1]
```diff
@@ -314,6 +314,15 @@ void validateLogEntry(LogEntryProto entry) {
 
   abstract CompletableFuture<Long> truncateImpl(long index);
 
+
+  @Override
+  public final CompletableFuture<Long> purge(long index) {
+    return runner.runSequentially(() -> purgeImpl(index));
```
Comment (andy2git) [inline ratis-server/src/main/java/org/apache/ratis/server/storage/RaftLog.java]: @szetszwo was wondering if need to run sequentially here?

**Comment by andy2git** [comment]
merged in 9dcb75a94eacae49b4fed7f7105b97deee4aba51. Close this request.

**Pull request closed without merging**